src/common/user-store/user-store.ts

```diff
@@ -165,55 +165,31 @@ export class UserStore extends BaseStore<UserStoreModel> /* implements UserStore
       this.lastSeenAppVersion = lastSeenAppVersion;
     }
 
-    this.httpsProxy = DESCRIPTORS.httpsProxy.fromStore(preferences?.httpsProxy);
-    this.shell = DESCRIPTORS.shell.fromStore(preferences?.shell);
-    this.colorTheme = DESCRIPTORS.colorTheme.fromStore(preferences?.colorTheme);
-    this.terminalTheme = DESCRIPTORS.terminalTheme.fromStore(preferences?.terminalTheme);
-    this.localeTimezone = DESCRIPTORS.localeTimezone.fromStore(preferences?.localeTimezone);
-    this.allowUntrustedCAs = DESCRIPTORS.allowUntrustedCAs.fromStore(preferences?.allowUntrustedCAs);
-    this.allowTelemetry = DESCRIPTORS.allowTelemetry.fromStore(preferences?.allowTelemetry);
-    this.allowErrorReporting = DESCRIPTORS.allowErrorReporting.fromStore(preferences?.allowErrorReporting);
-    this.downloadMirror = DESCRIPTORS.downloadMirror.fromStore(preferences?.downloadMirror);
-    this.downloadKubectlBinaries = DESCRIPTORS.downloadKubectlBinaries.fromStore(preferences?.downloadKubectlBinaries);
-    this.downloadBinariesPath = DESCRIPTORS.downloadBinariesPath.fromStore(preferences?.downloadBinariesPath);
-    this.kubectlBinariesPath = DESCRIPTORS.kubectlBinariesPath.fromStore(preferences?.kubectlBinariesPath);
-    this.openAtLogin = DESCRIPTORS.openAtLogin.fromStore(preferences?.openAtLogin);
-    this.hiddenTableColumns.replace(DESCRIPTORS.hiddenTableColumns.fromStore(preferences?.hiddenTableColumns));
-    this.syncKubeconfigEntries.replace(DESCRIPTORS.syncKubeconfigEntries.fromStore(preferences?.syncKubeconfigEntries));
-    this.editorConfiguration = DESCRIPTORS.editorConfiguration.fromStore(preferences?.editorConfiguration);
-    this.terminalCopyOnSelect = DESCRIPTORS.terminalCopyOnSelect.fromStore(preferences?.terminalCopyOnSelect);
-    this.terminalConfig = DESCRIPTORS.terminalConfig.fromStore(preferences?.terminalConfig);
-    this.updateChannel = DESCRIPTORS.updateChannel.fromStore(preferences?.updateChannel);
-    this.extensionRegistryUrl = DESCRIPTORS.extensionRegistryUrl.fromStore(preferences?.extensionRegistryUrl);
+    for (const [key, { fromStore }] of entries(DESCRIPTORS)) {
+      const curVal = this[key];
+      const newVal = fromStore((preferences)?.[key] as never) as never;
```

Comment:
is the `never` casting necessary? 

also couldn't this be just `preferences[key]`

Reply:
Yes unfortunately the casting is necessary.

Typescript has a limitation where doing this sort of thing in a loop isn't really supported by the compiler.